### PR TITLE
implemented DriverService,ManufacturerService

### PR DESCRIPTION
**Pull request by Daniel1004K**

### CODE REVIEW

**Review by Veronika-Alekseienko — APPROVED** [review]
Well done!

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            createDriverStatement.setString(1, driver.getName());
+            createDriverStatement.setString(2, driver.getLicenseNumber());
+            createDriverStatement.executeUpdate();
+            ResultSet resultSet = createDriverStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                driver.setId(resultSet.getObject(1, Long.class));
+            }
+            return driver;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create driver. " + driver + " ",
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:35]:
```suggestion
            throw new DataProcessingException("Couldn't create driver. " + driver,
```
let's do it everywhere

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            deleteDriverStatement.setLong(1, id);
+            return deleteDriverStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete a driver by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Long newId = resultSet.getObject("id", Long.class);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:111]:
```suggestion
        Long id = resultSet.getObject("id", Long.class);
```

File: src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private DriverDao driverDao;
+
+    @Override
+    public Driver create(Driver driver) {
+        return driverDao.create(driver);
+    }
+
+    @Override
+    public Driver get(Long id) {
+        return driverDao.get(id).orElse(null);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java:22]:
```suggestion
        return driverDao.get(id).orElseThrow()/get();
```

File: src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    private ManufacturerDao manufacturerDao;
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return manufacturerDao.create(manufacturer);
+    }
+
+    @Override
+    public Manufacturer get(Long id) {
+        return manufacturerDao.get(id).orElse(null);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java:22]: same here

**Review by Sofasmile — CHANGES_REQUESTED** [review]
great job)

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**